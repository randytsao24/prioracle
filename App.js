// App.js
// Entry point for Prioracle!

import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { Provider } from 'react-redux'
import { StackNavigator } from 'react-navigation';
import Login from './app/components/Login';
import ListingForm from './app/components/ListingForm';
import ProductAnalysis from './app/components/ProductAnalysis';
import { Icon, Header } from 'react-native-elements';
import store, { fetchListings } from  './app/store';
import AppHeader from './app/components/AppHeader';
import UserHome from './app/components/UserHome';
import UserListings from './app/components/UserListings';
import UserSettings from './app/components/UserSettings';
import Signup from './app/components/Signup';
import Intro from './app/components/Intro';

// Use this link to access our backend!
export const dbUrl = 'http://172.16.23.244:8080';
//export const dbUrl = 'http://192.168.1.4:8080';

export default class App extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    // Fetch all listings in our top-level component
    store.dispatch(fetchListings());
  }

  render() {
    return (
      <Provider store={store}>
        <RootNavigator />
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: { 
    flex: 1,
    justifyContent: 'space-between',
    backgroundColor: '#d14f4f'
  },
  headerContainer: {
    backgroundColor: '#d14f4f',
    padding: 20
  }
});

export const RootNavigator = StackNavigator({
  Main: {
    screen: Login,
    navigationOptions: {
      header: null
    }
  },
  UserHome: {
    screen: UserHome,
    navigationOptions: ({ navigation }) => ({
      title: 'Home',
      header: <AppHeader navigation={navigation} />
    })
  },
  ListingForm: {
    screen: ListingForm,
    navigationOptions: ({navigation}) => ({
      title: 'Add a New Listing',
      header: <AppHeader navigation={navigation} />
    })
  },
  Analysis: {
    screen: ProductAnalysis,
    path: 'listings/:id',
    navigationOptions: ({ navigation }) => ({
      title: 'Product Analysis',
      header: <AppHeader navigation={navigation} />
    })
  },
  UserListings: {
    path: 'user/:id/listings',
    screen: UserListings,
    navigationOptions: ({ navigation }) => ({
      title: 'Product Analysis',
      header: <AppHeader navigation={navigation} />
    })
  },
  Settings: {
    screen: UserSettings,
    navigationOptions: ({ navigation }) => ({
      title: 'Settings',
      header: <AppHeader navigation={navigation} />
    })
  },
  Signup: {
    screen: Signup,
    navigationOptions: {
      header: null
    }
  },
  Intro: {
    screen: Intro,
    navigationOptions: {
      header: null
    }
  }
});
