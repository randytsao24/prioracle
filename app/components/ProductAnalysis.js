// app/components/ProductAnalysis.js
// This component displays the price generated by our
// algorithms for the user to view.

import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, KeyboardAvoidingView } from 'react-native';
import { Button, ButtonGroup, FormLabel, FormInput, Card } from 'react-native-elements';
import { StackNavigator } from 'react-navigation';
import { connect } from 'react-redux';
import axios from 'axios';

import { dbUrl, RootNavigator } from '../../App';
import { fetchListings } from '../store';

// Options for the button group for editing or posting the listing
const optionGroup = ['Submit Listing', 'Edit'];

class ProductAnalysis extends Component {
  constructor(props) {
    super(props);

    this.state = {
      listing: {},
      selectedPrice: 0
    };
  }

  componentDidMount() {
    const listingId = Number(this.props.navigation.state.params.id);

    // Fetch listing associated with the listing ID param
    axios.get(dbUrl + `/api/listings/${listingId}`)
      .then(res => res.data)
      .then(listing => this.setState({
        listing,
        selectedPrice: 0.01 * listing.valuations[listing.valuations.length - 1].metaPrice
      }))
      .catch(error => console.log('Error:', error));
  }

  render() {
    const valuations = this.state.listing.valuations;

    return (
      <KeyboardAvoidingView 
        style={styles.container}
        behavior='padding' >
        <Card title={`Pricing analysis for ${this.state.listing && this.state.listing.name}`} >
          <Text>{'\n'}Prioracle has made a decision - a fair price for your product to sell at is ${`${valuations && valuations[valuations.length - 1].metaPrice}`} and the scraper price is {`${valuations && valuations[valuations.length - 1].scraperPrice}`}. You may modify the price below if you wish.{'\n'}</Text>
          <FormLabel labelStyle={styles.formLabel}>Suggested Price</FormLabel>
          <FormInput
            inputStyle={styles.inputText}
            textAlign={'center'}
            editable={true}
            value={this.state.selectedPrice.toString()}
            onChangeText={text => this.setState({selectedPrice: Number(text)})} />
        </Card>
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
    container: { 
        flex: 1,
        justifyContent: 'flex-start',
        backgroundColor: '#e1e8e6'
    },
    formLabel: {
      color: 'red'
    },
    inputContainer: {
      justifyContent: 'center'
    },
    inputText: {
      color: 'black'
    }
});

const mapStateToProps = (state) => {
  return {
    listings: state.listings.sort((a, b) => b.id - a.id)
  };
}

export default connect(mapStateToProps)(ProductAnalysis);
