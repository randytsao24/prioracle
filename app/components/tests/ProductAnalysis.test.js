// app/components/tests/ProductAnalysis.test.js
// Test basic functionality for our ProductAnalysis component

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../../../App';
import ProductAnalysis from '../ProductAnalysis';

configure({ adapter: new Adapter() });

const initialState = {
  listings: []
};

const navigation = {
  state: {
    params: {
      id: 1
    }
  }
};

const mockStore = configureStore();

test('ListingForm renders correctly', () => {
  const wrapper = shallow(
    <ProductAnalysis navigation={navigation} />,
    { context: { store: mockStore(initialState) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
  // const tree = renderer.create(<ListingForm />).toJSON();
  // expect(tree).toMatchSnapshot();
});
