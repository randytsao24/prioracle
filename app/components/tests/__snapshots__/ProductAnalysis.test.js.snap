// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductAnalysis
    dispatch={[Function]}
    listings={Array []}
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "id": 1,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "behavior": "padding",
      "children": <Card
        title="Pricing analysis for undefined"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          

          Prioracle has made a decision - a fair price for your product to sell at is $
          undefined
          . You may modify the price below if you wish.
          

        </Text>
        <FormLabel
          labelStyle={
            Object {
              "color": "red",
            }
          }
        >
          Suggested Price
        </FormLabel>
        <FormInput
          editable={true}
          inputStyle={
            Object {
              "color": "black",
            }
          }
          normalizeFontSize={true}
          onChangeText={[Function]}
          textAlign="center"
          value="0"
        />
      </Card>,
      "keyboardVerticalOffset": 0,
      "style": Object {
        "backgroundColor": "#e1e8e6",
        "flex": 1,
        "justifyContent": "flex-start",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            

            Prioracle has made a decision - a fair price for your product to sell at is $
            undefined
            . You may modify the price below if you wish.
            

          </Text>,
          <FormLabel
            labelStyle={
              Object {
                "color": "red",
              }
            }
          >
            Suggested Price
          </FormLabel>,
          <FormInput
            editable={true}
            inputStyle={
              Object {
                "color": "black",
              }
            }
            normalizeFontSize={true}
            onChangeText={[Function]}
            textAlign="center"
            value="0"
          />,
        ],
        "title": "Pricing analysis for undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "
",
              "Prioracle has made a decision - a fair price for your product to sell at is $",
              "undefined",
              ". You may modify the price below if you wish.",
              "
",
            ],
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": Array [
            "
",
            "Prioracle has made a decision - a fair price for your product to sell at is $",
            "undefined",
            ". You may modify the price below if you wish.",
            "
",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Suggested Price",
            "labelStyle": Object {
              "color": "red",
            },
          },
          "ref": null,
          "rendered": "Suggested Price",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "editable": true,
            "inputStyle": Object {
              "color": "black",
            },
            "normalizeFontSize": true,
            "onChangeText": [Function],
            "textAlign": "center",
            "value": "0",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "behavior": "padding",
        "children": <Card
          title="Pricing analysis for undefined"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            

            Prioracle has made a decision - a fair price for your product to sell at is $
            undefined
            . You may modify the price below if you wish.
            

          </Text>
          <FormLabel
            labelStyle={
              Object {
                "color": "red",
              }
            }
          >
            Suggested Price
          </FormLabel>
          <FormInput
            editable={true}
            inputStyle={
              Object {
                "color": "black",
              }
            }
            normalizeFontSize={true}
            onChangeText={[Function]}
            textAlign="center"
            value="0"
          />
        </Card>,
        "keyboardVerticalOffset": 0,
        "style": Object {
          "backgroundColor": "#e1e8e6",
          "flex": 1,
          "justifyContent": "flex-start",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              

              Prioracle has made a decision - a fair price for your product to sell at is $
              undefined
              . You may modify the price below if you wish.
              

            </Text>,
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Suggested Price
            </FormLabel>,
            <FormInput
              editable={true}
              inputStyle={
                Object {
                  "color": "black",
                }
              }
              normalizeFontSize={true}
              onChangeText={[Function]}
              textAlign="center"
              value="0"
            />,
          ],
          "title": "Pricing analysis for undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "
",
                "Prioracle has made a decision - a fair price for your product to sell at is $",
                "undefined",
                ". You may modify the price below if you wish.",
                "
",
              ],
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": Array [
              "
",
              "Prioracle has made a decision - a fair price for your product to sell at is $",
              "undefined",
              ". You may modify the price below if you wish.",
              "
",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Suggested Price",
              "labelStyle": Object {
                "color": "red",
              },
            },
            "ref": null,
            "rendered": "Suggested Price",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "editable": true,
              "inputStyle": Object {
                "color": "black",
              },
              "normalizeFontSize": true,
              "onChangeText": [Function],
              "textAlign": "center",
              "value": "0",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
